lib/data_stores/mysql_advanced_data_store.py

```diff
@@ -26,23 +26,48 @@ class Error(data_store.Error):
   """Base class for all exceptions in this module."""
 # pylint: enable=nonstandard-exception
 
+class SafeQueue(Queue.Queue):
+  def __init__(self, maxsize=0):
```

Comment:
Yeah you're right. I came up with a few alternatives but none of them are particularly good.  Discussing with andy, we'll figure out what to do on monday.